modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/checkpoint/CheckpointReadWriteLock.java

```diff
@@ -182,14 +199,35 @@ public boolean hasQueuedWriters() {
     }
 
     private void onReadLock(long start, boolean taken) {
-        long elapsed = coarseCurrentTimeMillis() - start;
+        readLockWaitingThreadsGauge.decrementAndGet();
+
+        long coarseCurrentTimeMillis = coarseCurrentTimeMillis();
```

Comment:
`coarseCurrentTimeMillis()` has an abysmal precision. I would recommend replacing it with `nanoTime()` as the best tool for measuring time. There's also `org.apache.ignite.internal.util.IgniteUtils#monotonicMs` which hides some complexity, but adds some extra CPU instructions